tui-markdown/src/options.rs

```diff
@@ -0,0 +1,40 @@
+//! Rendering configuration for tui-markdown.
+//!
+//! For now the only knob is the theme [`StyleSheet`].  The struct purposefully
+//! remains `#[non_exhaustive]` *in spirit* (not in syntax to avoid MSRV
+//! bumps): we recommend construction through the builder pattern or the field
+//! shorthand with `..Default::default()` so that adding new options becomes a
+//! non-breaking change.
+
+use crate::style_sheet::{BuiltinStyleSheet, StyleSheet};
+
+/// Collection of optional parameters that influence markdown rendering.
+///
+/// The generic `S` allows users to carry their own theme type without forcing
+/// dynamic dispatch.  Libraries that do not want the extra generic in their
+/// call-chain can always box it (`Box<dyn StyleSheet>`).
+#[derive(Clone)]
+pub struct Options<S: StyleSheet = BuiltinStyleSheet> {
+    /// The [`StyleSheet`] implementation that will be consulted every time the
+    /// renderer needs a colour choice.
+    pub styles: S,
+
+    /// Wrap text at this many columns.  `None` means *no wrapping* (current
+    /// behaviour).
```

Comment:
Current behavior refers to before the Option type was introduced, but won't have any relevance to someone who's looking at the docs for the first time, so it doesn't need to be stated.

```suggestion
    /// Wrap text at this many columns.
    ///
    /// The default value is `None` which indicates that there is no wrapping.
```